core/pkg/server/server.go

```diff
@@ -55,9 +60,35 @@ func NewServer(ctx context.Context, addr string, portFile string) (*Server, erro
 
 	s.wg.Add(1)
 	go s.Serve()
```

Comment:
I clean up the server api a bit here: https://github.com/wandb/wandb/pull/7526, let's merge it first? otherwise, for example, our constructor does more than it's supposed to.

Reply:
i merged with your code, much cleaner.